doc/manual/manual.md

```diff
@@ -2476,3 +2476,249 @@ Currently, this function can only be used for LLVM verification, though
 that will likely be generalized in the future. It can be used in either
 the pre state or the post state, to specify the value of ghost state
 either before or after the execution of the function, respectively.
+
+## An Extended Example
+
+To tie together many of the concepts in this manual, we now present a
+non-trivial verification task in its entirety. All of the code for this example
+can be found in the `examples/salsa20` directory of
+[the SAWScript repository](https://github.com/GaloisInc/saw-script).
+
+### Salsa20 Overview
+
+Salsa20 is a stream cipher developed in 2005 by Daniel J. Bernstein, built on a
+pseudorandom function utilizing add-rotate-XOR (ARX) operations on 32-bit
+words[^4]. Bernstein himself has provided several public domain
+implementations of the cipher, optimized for common machine architectures.
+For the mathematically inclined, his specification for the cipher can be
+found [here](http://cr.yp.to/snuffle/spec.pdf).
+
+The repository referenced above contains three implementations of the Salsa20
+cipher: A reference Cryptol implementation (which we take as correct in this
+example), and two C implementations, one of which is from Bernstein himself.
+For this example, we focus on the second of these C
+implementations, which more closely matches the Cryptol implementation. Full
+verification of Bernstein's implementation is available in
+`examples/salsa20/djb`, for the interested. The code for this verification task
+can be found in the files named according to the pattern
+`examples/salsa20/(s|S)alsa20.*`.
+
+### Specifications
+
+We now take on the actual verification task. This will be done in two stages:
+We first define some useful utility functions for constructing common patterns
+in the specifications for this type of program (i.e. one where the arguments to
+functions are modified in-place.) We then demonstrate how one might construct a
+specification for each of the functions in the Salsa20 implementation described
+above.
+
+#### Utility Functions
+
+We first define the function
+`alloc_init : LLVMType -> Term -> CrucibleSetup SetupValue`.
+
+`alloc_init ty v` returns a `SetupValue` consisting of a pointer to memory
+allocated and initialized to a value `v` of type `ty`. `alloc_init_readonly`
+does the same, except the memory allocated cannot be written to.
+
+~~~~
+import "Salsa20.cry";
+
+let alloc_init ty v = do {
+    p <- crucible_alloc ty;
+    crucible_points_to p (crucible_term v);
+    return p;
+};
+
+let alloc_init_readonly ty v = do {
+    p <- crucible_alloc_readonly ty;
+    crucible_points_to p (crucible_term v);
+    return p;
+};
+~~~~
+
+We now define
+`ptr_to_fresh : String -> LLVMType -> CrucibleSetup (Term, SetupValue)`.
+
+`ptr_to_fresh n ty` returns a pair `(x, p)` consisting of a fresh symbolic
```

Comment:
I'm not thrilled with starting a sentence with verbatim text, but sometimes avoiding it leads to awkward sentences.